packages/app/src/app/overmind/namespaces/dashboard/actions.ts

```diff
@@ -683,25 +774,70 @@ export const renameFolderInState: Action<{ path: string; newPath: string }> = (
   dashboard.allCollections = newFolders;
 };
 
+export const renameRepoSandboxInState: Action<{
+  id: string;
+  title: string;
+}> = ({ state }, { id, title }) => {
+  if (
+    !state.dashboard.sandboxes.REPOS ||
+    !state.dashboard.sandboxes.REPOS[repoName]
+  ) {
+    return;
+  }
+
+  const repoSandboxes = state.dashboard.sandboxes.REPOS[repoName];
+  state.dashboard.sandboxes.REPOS = {
+    ...state.dashboard.sandboxes.REPOS,
+    [repoName]: {
+      ...repoSandboxes,
+      sandboxes: repoSandboxes?.sandboxes.map(sandbox => {
+        if (sandbox.id === id) {
+          return {
+            ...sandbox,
+            title,
+          };
+        }
+
+        return sandbox;
+      }),
+    },
+  };
+};
+
 export const renameSandbox: AsyncAction<{
   id: string;
   title: string;
   oldTitle: string;
 }> = async ({ effects, actions }, { id, title, oldTitle }) => {
-  actions.dashboard.renameSandboxInState({
-    id,
-    title,
-  });
-  try {
-    await effects.gql.mutations.renameSandbox({
+  if (isRepoPage) {
+    actions.dashboard.renameRepoSandboxInState({
```

Comment:
Same for here, shouldn't we handle renaming repo sandboxxes in `renameSandboxInState`. I'd like to know your opinion on this as well, @christianalfoni.

An idea that I had before would be that we actually keep all sandboxes in one array, and create `derived` functions for ordering them in `Recents`, `AllSandboxes`, `Repos`, etc... That would allow us to eg. only rename one instance of a sandbox and it's reflected everywhere immediately, also would save a lot of work when we're going to add live updates. I realize that this should be a PR on its own, but I'm thinking that if we're going to plan to do this anyway, then it would be better if we already put the renaming logic in a single function (`renameRepoSandboxInState`) instead of calling it conditionally.

Reply:
this is done, I don't know what the rest is ahah